cfme/tests/infrastructure/test_snapshot.py

```diff
@@ -174,6 +174,47 @@ def test_verify_revert_snapshot(full_test_vm, provider, soft_assert, register_ev
 
 
 @pytest.mark.uncollectif(lambda provider: not provider.one_of(VMwareProvider))
+def test_operations_suspended_vm(small_test_vm, provider, soft_assert):
+    """Tests snapshot operations on suspended vm
+
+    Metadata:
+        test_flag: snapshot, provision
+    """
+    snapshot1 = new_snapshot(small_test_vm)
+    snapshot1.create()
+    wait_for(lambda: snapshot1.active, num_sec=300, delay=20, fail_func=snapshot1.refresh)
+    small_test_vm.power_control_from_cfme(option=small_test_vm.SUSPEND, cancel=False)
+    small_test_vm.wait_for_vm_state_change(desired_state=small_test_vm.STATE_SUSPENDED)
+    current_state = small_test_vm.find_quadicon().state
+    soft_assert(current_state.startswith('currentstate-suspended'),
+                "Quadicon state is {}".format(current_state))
+    snapshot2 = new_snapshot(small_test_vm)
+    snapshot2.create()
+    wait_for(lambda: snapshot2.active, num_sec=300, delay=20, fail_func=snapshot2.refresh)
+    snapshot1.revert_to()
+    wait_for(lambda: snapshot1.active, num_sec=300, delay=20, fail_func=snapshot1.refresh)
+    current_state = small_test_vm.find_quadicon().state
+    soft_assert(current_state.startswith('currentstate-off'),
+                "Quadicon state is {}".format(current_state))
+    assert small_test_vm.provider.mgmt.is_vm_stopped(small_test_vm.name)
+    snapshot2.revert_to()
+    wait_for(lambda: snapshot2.active, num_sec=300, delay=20, fail_func=snapshot2.refresh)
+    current_state = small_test_vm.find_quadicon().state
+    soft_assert(current_state.startswith('currentstate-suspended'),
+                "Quadicon state is {}".format(current_state))
+    assert small_test_vm.provider.mgmt.is_vm_suspended(small_test_vm.name)
+    snapshot2.delete_all()
+    try:
+        wait_for(lambda: not snapshot1.exists, num_sec=300, delay=20, fail_func=snapshot1.refresh)
```

Comment:
You can use ``fail_condition=True`` to avoid lambda

Reply:
I didn't realized that it's a property, not function, so disregard my comment :)